File: (7 kyu) Simple beads count/(7 kyu) Simple beads count.js
// #1
// function countRedBeads(n) {
//   return n < 2 ? 0 : (n - 1) * 2;
// }
// #2
// const countRedBeads = (n) => (n < 2 ? 0 : (n - 1) * 2);
// #3
// const countRedBeads = (n) => ((n || 1) - 1) * 2;
// #4
// maximum value -> two for every two beads times number of blue beads - 1
const countRedBeads = (n) => Math.max(0, (n - 1) * 2);

File: (7 kyu) Simple beads count/(7 kyu) Simple beads count.java
// 1 Plain solution
public class BeadsCounter {
  public static int countRedBeads(final int nBlue) {
    if (nBlue < 2) {
      return 0;
    } else {
      return (nBlue - 1) * 2;
    }
  }
}

// 2 Optimized solution
public class BeadsCounter {
  public static int countRedBeads(final int nBlue) {
    return nBlue < 2 ? 0 : (nBlue - 1) * 2;
  }
}

// 3 Clever solution
public class BeadsCounter {
  public static int countRedBeads(final int nBlue) {
    return Math.max(0, (nBlue - 1) * 2);
  }
}
